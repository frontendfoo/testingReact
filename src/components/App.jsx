import React, { memo } from 'react'

const App = ({ handleSetName, name, disabled, handleSubmitName }) => (
  <div>
    <h1>login</h1>
    <input
      type='text'
      className='w-full mt-8'
      placeholder='Enter your name here...'
      onChange={handleSetName}
      value={name}
    />
    <div>
      <button className='mt-12' disabled={disabled} onClick={handleSubmitName}>
        Login!
      </button>
    </div>
  </div>
)
export default memo(App)
