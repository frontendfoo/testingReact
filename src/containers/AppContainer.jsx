import React from 'react'
import App from '../components/App'

// Custom Hooks
import useLoginForm from '../hooks/useLoginForm'

const AppContainer = () => {
  const { handleSetName, name, disabled, handleSubmit } = useLoginForm()

  const appProps = {
    handleSetName,
    name,
    disabled,
    handleSubmit,
  }

  return <App {...appProps} />
}
export default AppContainer
