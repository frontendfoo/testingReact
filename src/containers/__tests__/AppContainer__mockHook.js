// Test Renderer
import TestRenderer from 'react-test-renderer'

// React
import React from 'react'

// Container & Component
import AppContainer from '../AppContainer'
import App from '../../components/App'

jest.mock('../../components/App', () => ({
  __esModule: true,
  default: function mockComponent(props) {
    return <div {...props} />
  },
}))

jest.mock('../../hooks/useLoginForm')

describe('The App Container Component', () => {
  it('should map the props to App component', () => {
    const { root } = TestRenderer.create(<AppContainer />)
    const props = root.findByType(App).props
    expect(props).toMatchSnapshot()
  })
})
