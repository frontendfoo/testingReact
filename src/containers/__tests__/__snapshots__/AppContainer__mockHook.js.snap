// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The App Container Component should map the props to App component 1`] = `
Object {
  "disabled": false,
  "handleSetName": [MockFunction handleSetName],
  "handleSubmit": [MockFunction handleSubmit],
  "name": "Test Name",
}
`;
